src/Core/Checkout/Cart/Rule/LineItemCustomFieldRule.php

```diff
@@ -109,4 +109,61 @@ private function isCustomFieldValid(LineItem $lineItem): bool
                 throw new UnsupportedOperatorException($this->operator, self::class);
         }
     }
+
+    /**
+     * @return Constraint[]
+     */
+    private function getRenderedFieldValueConstraints(): array
+    {
+        $constraints = [];
+
+        if (!\is_array($this->renderedField) || !\array_key_exists('type', $this->renderedField)) {
+            return [new NotBlank()];
+        }
+
+        switch ($this->renderedField['type']) {
```

Comment:
switch case needed?

Reply:
I personally prefer switch cases in situations where I change behaviour depending on one given field. If you prefer if/else I can certainly change that, just give me the word :-)

Reply:
I don't see the need if there is only one case 😉

Reply:
also switch case is not type safe 😉

Reply:
Alright, I changed all switch-cases to if-statements.